### DISCUSSION TITLE
Kubernetes mixin usage

### DISCUSSION THREAD

**Discussion by dejanu** [getporter/porter#3191] [Q&A] [answered]
From what I understand, the Kubernetes mixin uses `kubectl apply` and `kubectl delete` for install/upgrade/uninstall actions. I am curious about the recommended approach for patching resource i.e. `kubectl patch`. Thanks!

**Comment by kichristensen** [comment]
Sorry for the late reply. 
Currently the Kubernetes mixin doesn't support patch, but I would be very interested in understanding your use case in more detail.

**Reply by dejanu** [reply]
I would like to patch a service. In the manifest the svc is ClusterIP, but I would like to provide the possibility to the bundle's user to have an action that can patch that service to a LoadBalancer.  
Or maybe there's another way to do it ?  i.e.  a parameter that the user can pass at runtime when the action is invoked ?
Thx

**Reply by kichristensen** [reply, marked as answer]
Not saying that it wouldn't make sense to support patch in the mixin, but it sounds like a good case for Kustomize or Helm, both of them have mixins available. Any reason you are not using those?

**Reply by dejanu** [reply]
Not really, I’m just beginning to grasp how to "combine" mixins, I had tunnel vision 😄
